In [2]:
import numpy as np
import matplotlib.pyplot as plt

In [144]:
# Me estaba dando muchos problemas para cargar los datos por las comas entre algunos de los valores de la columna grantee
# No he encontrado una manera mejor asi que he abierto los datos en excel y le he cambiado el separador a ";" en lugar de "," 
# Asi almenos he coneguido cargar los datos con numpy

especulaciondf = np.genfromtxt("D:\Documentos\Master\R\datos\Especulacion.csv",  delimiter = ";", dtype = str)


In [145]:
colnames = especulaciondf[0]
colnames

# Creamos un array con solo los precios, desde la fila 1 para no incluir la cabecera y aplicamos transformaciones para quitar
# las comas que dan problemas y convertir los datos en tipo int para poder calcular.

price = especulaciondf[1:,11]
price = [i.replace(",","") for i in price]
price = np.asarray(price)
price = price.astype(int)


np.mean(price)
np.max(price)
np.min(price)

500000

In [146]:
np.mean(price)

9284909.485018726

In [147]:
np.max(price)

90000000

In [148]:
np.min(price)

500000

In [149]:
colnames

#Hacemos los mismo para ratio de capitalización

cap_rate = especulaciondf[1:,12]
cap_rate = [i.replace(",","") for i in cap_rate]
cap_rate = np.asarray(cap_rate)
cap_rate = cap_rate.astype(float)


In [150]:
np.mean(cap_rate)

0.03393258426966293

In [151]:
np.max(cap_rate)

0.068

In [152]:
np.min(cap_rate)

0.008

El ratio de capitalización es el "net_operating_income/Property_value" lo que quiere decir los ingresos que genera una propiedad proporcionales a su valor.

Vemos que el cap_rate medio es del 3,3%, el máximo 6,8% y el mas bajo 0,8%

In [153]:
postcode = especulaciondf[1:,15]
postcode

# Ahora vamos a encontrar cuantas propiedades hay por codigo postal

#Con este codigo contamos los valores unicos del array de la columna postcode y sacamos sus indices

valores_unicos , indices = np.unique(postcode, return_inverse=True)

# Con un bucle guardamos en arrays diferentes cada vez que aparezca el mismo valor
agrupacion = [postcode[indices == i] for i in range(len(valores_unicos))]

# Y finalmente contamos cuantos elementos hay en cada array creado con la funcion len que nos da la longitud del array

conteo = [len(unico) for unico in agrupacion]

In [154]:
# Añadimos el valor unico al lado de cada conteo porque si no se entiende nada.

listado_final = list(zip(valores_unicos, conteo))
listado_final

[('', 5),
 ('10002', 11),
 ('10003', 11),
 ('10009', 8),
 ('10011', 7),
 ('10012', 3),
 ('10013', 5),
 ('10014', 1),
 ('10019', 3),
 ('10021', 2),
 ('10022', 3),
 ('10023', 1),
 ('10024', 6),
 ('10025', 6),
 ('10026', 12),
 ('10027', 4),
 ('10029', 2),
 ('10030', 2),
 ('10031', 13),
 ('10032', 21),
 ('10033', 24),
 ('10034', 2),
 ('10035', 1),
 ('10036', 1),
 ('10038', 1),
 ('10040', 5),
 ('10065', 3),
 ('10075', 1),
 ('10128', 5),
 ('10301', 3),
 ('10302', 1),
 ('10451', 2),
 ('10452', 9),
 ('10453', 9),
 ('10454', 5),
 ('10455', 5),
 ('10456', 9),
 ('10457', 7),
 ('10458', 24),
 ('10459', 7),
 ('10460', 5),
 ('10461', 11),
 ('10462', 8),
 ('10463', 4),
 ('10466', 2),
 ('10467', 10),
 ('10468', 17),
 ('10469', 2),
 ('10470', 1),
 ('10471', 1),
 ('10472', 6),
 ('10474', 1),
 ('11101', 2),
 ('11102', 2),
 ('11103', 5),
 ('11105', 3),
 ('11106', 2),
 ('11201', 1),
 ('11203', 5),
 ('11204', 5),
 ('11206', 1),
 ('11207', 1),
 ('11208', 1),
 ('11209', 9),
 ('11210', 7),
 ('11211', 5),
 ('11

Obtenemos cuantas observaciones hay por cada postcode, en este caso casas diferentes por cada codigo postal.

Vemos que el codigo postal 10458 junto con el 10033 con 24 propiedades son los que mas. Buscando en Google sale que son los barrios de Belmont|The Bronx y Washington Heights respectivamente en el estado de Nueva York.